DeveloperDocs.md

```diff
@@ -7,25 +7,95 @@
+4. Install Docker
+5. Install postgresql
+6. Run a postgresql server in a docker container:
+
+`docker run --name dt_dev -e POSTGRES_PASSWORD=password -p 5432:5432 -d postgres`
+
+7. To run the backend, we first need to create an empty database:
+
+`createdb --host localhost --username postgres dt_dev`
+
+#### Running the tests
+
+1. Tests can now be run by locally by navigating into dtbase: `cd dtbase` and running the tests: `pytest tests`
+
+#### Running the backend API
+
+The backend API is a flask app that provides REST API endpoints to facilitate reading and writing to the database.
+1. Navigate to the directory `dtbase/backend` and run the command `./run_localdb.sh`. You should then have the flask app listening on `http://localhost:5000` and be able to send HTTP requests to it.  See the [API docs](dtbase/backend/README.md) for details.
+
+#### Running the frontend API
+
+The DTBase frontend is currently an extremely lightweight Flask webapp:
+1. Install npm
+2. Navigate to the directory `dtbase/backend` and run the command `./run.sh`.
+3. You should then have the flask app listening on `http://localhost:5000` and be able to send HTTP requests to it.  See the [API docs](dtbase/backend/README.md) for details.
+
+
+### Running an Azure Deployed Instance of DTBase
+
+1. Copy the file `.secrets/dtenv_template.sh` to `.secrets/dtenv.sh` and populate this file with values for the various environment variables (ask an existing developer for these). These environment variables will point towards an already existing database hosted on Azure.
+
+#### Running the backend API
 
 The backend API is a flask app that provides REST API endpoints to facilitate reading and writing to the database.
-* Navigate to the directory `dtbase/backend`.
-* Run the command `./run.sh` if you will be using a cloud-based database, or `./run_localdb.sh` if you will be using a docker postgres container on your local machine.
-* You should then have the flask app listening on `http://localhost:5000` and be able to send HTTP requests to it.  See the [API docs](dtbase/backend/README.md) for details.
 
-### Running the frontend
+1. Your IP Address must be whitelisted on Azure. Ask one of the developers to help with this.
+2. Navigate to the directory `dtbase/backend` and Run the command `./run.sh`.
 
-The DTBase frontend is currently an extremely lightweight Flask webapp.   To run this:
-* Navigate to the directory `dtbase/webapp`.
-* Set the environment variable `DT_BACKEND_URL` to point to the backend, i.e. if following the instructions above, do
+You should then have the flask app listening on `http://localhost:5000` and be able to send HTTP requests to it.  See the [API docs](dtbase/backend/README.md) for details.
+
+#### Running the frontend
+
+The DTBase frontend is currently an extremely lightweight Flask webapp:
+1.  Navigate to the directory `dtbase/webapp`.
+2. Set the environment variable `DT_BACKEND_URL` to point to the backend:
```

Comment:
Is this also necessary when doing the fully local deployment? Actually, would we be able to say that running the front end is the same regardless of whether the database is remote or local?